satpy/readers/seviri_l2_bufr.py

```diff
@@ -171,30 +171,29 @@ def get_array(self, key):
         return arr
 
     def get_dataset(self, dataset_id, dataset_info):
-        """Get dataset using the BUFR key in dataset_info."""
+        """Get dataset using the BUFR key in dataset_info. The create dataset with or without an AreaDefinition."""
         arr = self.get_array(dataset_info['key'])
-        arr[arr == dataset_info['fill_value']] = np.nan
 
-        if not self.with_adef:
-            xarr = self.get_dataset_with_swath_def(arr, dataset_info)
+        if self.with_adef:
+            xarr = self.get_dataset_with_area_def(arr, dataset_id)
+            # coordinates are not relevant when returning data with an AreaDefinition
+            if 'coordinates' in dataset_info.keys():
+                del dataset_info['coordinates']
         else:
-            xarr = self.get_dataset_with_area_def(arr, dataset_id, dataset_info)
+            xarr = xr.DataArray(arr, dims=["y"])
 
-        return xarr
+        if 'fill_value' in dataset_info:
+            xarr = xarr.where(xarr != dataset_info['fill_value'], np.nan)
```

Comment:
The default for this method is to fill with `np.nan` already...

Reply:
True, fixed now.